### PR TITLE
Base container

### PR DESCRIPTION
**Pull request by drivera-armedia**
First version of the container to be merged to the main branch

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,27 @@
+ARG ARCH="amd64"
+ARG OS="linux"
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
+LABEL maintainer="Armedia, LLC"
```
Comment (fabricetriboix) [inline Dockerfile]: Add more labels

File: Dockerfile [hunk 2]
```diff
@@ -1 +1,27 @@
...
 
+ARG ARCH="amd64"
+ARG OS="linux"
```
Comment (fabricetriboix) [inline Dockerfile:7]: Duplicate arguments

File: Dockerfile [hunk 3]
```diff
@@ -1 +1,27 @@
...
+ARG VER="1.4.1"
+ARG PKG="pushgateway"
+ARG SRC="${PKG}-${VER}.${OS}-${ARCH}"
+
+RUN curl \
+		-L "https://github.com/prometheus/${PKG}/releases/download/v${VER}/${SRC}.tar.gz" \
+		-o package.tar.gz && \
```
Comment (fabricetriboix) [inline Dockerfile]: Please use S3 bucket and `get-artifcats.sh` script

File: Dockerfile [hunk 4]
```diff
@@ -1 +1,27 @@
...
+	tar -xzvf "package.tar.gz" && \
+	mkdir -p "/pushgateway" && \
+	mv -vif "${SRC}/LICENSE"                     "/LICENSE" && \
+	mv -vif "${SRC}/NOTICE"                      "/NOTICE" && \
+	mv -vif "${SRC}/pushgateway"                 "/bin/pushgateway" && \
+	chown nobody:nobody "/bin/pushgateway" "/pushgateway" && \
```
Comment (fabricetriboix) [inline Dockerfile]: Please use `root:root`

File: Dockerfile [hunk 5]
```diff
@@ -1 +1,27 @@
...
+	rm -rvf "${SRC}" package.tar.gz
+
+USER nobody
```
Comment (fabricetriboix) [inline Dockerfile]: Please use `useradd` to create a specific user

File: Dockerfile [hunk 6]
```diff
@@ -1 +1,27 @@
...
+EXPOSE 9091
+WORKDIR /pushgateway
+
+ENTRYPOINT [ "/bin/pushgateway" ]
```
Comment (fabricetriboix) [inline Dockerfile]: Please replace `ENTRYPOINT` by `CMD`

**Review by fabricetriboix — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 7]
```diff
@@ -1 +1,32 @@
...
+
+LABEL	ORG="Armedia LLC" \
+		APP="Prometheus Push Gateway" \
+		VERSION="${VER}" \
+		IMAGE_SOURCE="https://github.com/ArkCase/ark_prometheus_pushgate" \
+		MAINTAINER="Armedia LLC"
+
+# Modify to fetch from S3 ...
+RUN curl \
+		-L "https://github.com/prometheus/${PKG}/releases/download/v${VER}/${SRC}.tar.gz" \
```
Comment (JonHolman) [inline Dockerfile]: add get artifacts script, put this file in our artifacts bucket

File: Dockerfile [hunk 8]
```diff
@@ -1 +1,32 @@
...
+	tar -xzvf "package.tar.gz" && \
+	mkdir -pv "/app/data" && \
+	mv -vif "${SRC}/LICENSE"					"/LICENSE" && \
+	mv -vif "${SRC}/NOTICE"						"/NOTICE" && \
+	mv -vif "${SRC}/pushgateway"				"/bin/pushgateway" && \
+	chown -R nobody:nobody	"/app/data" && \
+	chmod -R ug+rwX,o-rwx	"/app/data" && \
+	rm -rvf "${SRC}" "package.tar.gz"
+
+USER		nobody
```
Comment (JonHolman) [inline Dockerfile]: As @fabricetriboix said on the other PR, please don't use nobody.  Create a user and use it.

File: Dockerfile [hunk 9]
```diff
@@ -1 +1,32 @@
...
+EXPOSE		9091
+VOLUME		[ "/app/data" ]
+WORKDIR		/app/data
+ENTRYPOINT	[ "/bin/pushgateway" ]
```
Comment (JonHolman) [inline Dockerfile]: As @fabricetriboix said on the other PR, please use CMD not ENTRYPOINT

File: Dockerfile [hunk 10]
```diff
@@ -1 +1,32 @@
...
+	mv -vif "${SRC}/NOTICE"						"/NOTICE" && \
+	mv -vif "${SRC}/pushgateway"				"/bin/pushgateway" && \
+	chown -R nobody:nobody	"/app/data" && \
+	chmod -R ug+rwX,o-rwx	"/app/data" && \
+	rm -rvf "${SRC}" "package.tar.gz"
+
+USER		nobody
+EXPOSE		9091
+VOLUME		[ "/app/data" ]
+WORKDIR		/app/data
```
Comment (JonHolman) [inline Dockerfile]: as @fabricetriboix said on the other PR, is this right?  Volume === WORKDIR?

**Review by JonHolman — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 11]
```diff
@@ -1,27 +1,32 @@
...
 ARG PKG="pushgateway"
 ARG SRC="${PKG}-${VER}.${OS}-${ARCH}"
 
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
+
+LABEL	ORG="Armedia LLC" \
+		APP="Prometheus Push Gateway" \
+		VERSION="${VER}" \
+		IMAGE_SOURCE="https://github.com/ArkCase/ark_prometheus_pushgate" \
+		MAINTAINER="Armedia LLC"
```
Comment (fabricetriboix) [inline Dockerfile]: Inconsistent with ark_prometheus

**Review by fabricetriboix — DISMISSED** [review]

**Review by fabricetriboix — APPROVED** [review]

**Review by JonHolman — APPROVED** [review]

**Review by JonHolman — APPROVED** [review]

**Pull request merged**